ocaml/sdk-gen/common/CommonFunctions.ml

```diff
@@ -111,137 +92,180 @@ and is_method_static message =
   | {param_type= ty; _} :: _ ->
       not (ty = Ref message.msg_obj_name)
 
-and get_method_params_list message =
-  if is_method_static message then
-    message.msg_params
-  else
-    List.tl message.msg_params
+and is_setter message = String.starts_with ~prefix:"set" message.msg_name
 
-and code_name_of_release_ext x =
-  try code_name_of_release x with UnspecifiedRelease -> ""
+and is_getter message = String.starts_with ~prefix:"get" message.msg_name
 
-and code_name_order = release_order |> List.map code_name_of_release
+and is_adder message = String.starts_with ~prefix:"add" message.msg_name
 
-and gen_param_groups message params =
-  let expRelease = get_prototyped_release message.msg_lifecycle in
-  let msgRelease = get_first_release message.msg_release.internal in
-  let msgReleaseIndex = list_index_of msgRelease code_name_order in
-  let paramGroups = gen_param_groups_for_releases code_name_order params in
-  let rec getValid x =
-    match x with
-    | [] ->
-        []
-    | hd :: tl ->
-        let index = list_index_of (fst hd) code_name_order in
-        let valid =
-          if
-            (not (index = -1))
-            && (not (msgReleaseIndex = -1))
-            && index < msgReleaseIndex
-          then
-            []
-          else
-            snd hd
-        in
-        valid :: getValid tl
-  in
-  let filteredGroups =
+and is_remover message = String.starts_with ~prefix:"remove" message.msg_name
+
+and is_constructor message =
+  message.msg_tag = FromObject Make || message.msg_name = "create"
+
+and is_real_constructor message = message.msg_tag = FromObject Make
+
+and is_destructor message =
+  message.msg_tag = FromObject Delete || message.msg_name = "destroy"
+
+let code_name_order = release_order |> List.map code_name_of_release
+
+let compare_versions x y =
```

Comment:
This could be made simpler by relying on the natural order of tuples.
```
let compare x y =
  let (x1, x2, x3 as xx) = x.major, x.minor, x.micro in
  let (y1, y2, y3 as yy) = y.major, y.minor, y.micro in
    compare xx yy
```